hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -379,21 +381,19 @@ func (r *AzureDeploymentReconciler) StartDeleteOfResource(ctx context.Context) (
 		return ctrl.Result{}, errors.Wrapf(err, "couldn't convert to armResourceSpec")
 	}
 
-	var retryAfter time.Duration // ARM can tell us how long to wait for a DELETE
-
-	err = r.Patch(ctx, func(ctx context.Context, mutData *AzureDeploymentReconciler) error {
-		emptyStatus, deleteErr := reflecthelpers.NewEmptyArmResourceStatus(mutData.obj)
-		if deleteErr != nil {
-			return errors.Wrapf(deleteErr, "creating empty status for %q", resource.GetId())
-		}
+	emptyStatus, err := reflecthelpers.NewEmptyArmResourceStatus(r.obj)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "creating empty status for %q", resource.GetId())
+	}
 
-		retryAfter, deleteErr = r.ARMClient.BeginDeleteResource(ctx, resource.GetId(), resource.Spec().GetApiVersion(), emptyStatus)
-		if deleteErr != nil {
-			return errors.Wrapf(deleteErr, "deleting resource %q", resource.Spec().GetType())
-		}
+	// retryAfter = ARM can tell us how long to wait for a DELETE
+	retryAfter, err := r.ARMClient.BeginDeleteResource(ctx, resource.GetId(), resource.Spec().GetApiVersion(), emptyStatus)
```

Comment:
We don't seem to ever use `retryAfter`

Reply:
it is used in the return value down below (if it wasn't used at all then the linter would complain!)

Reply:
I searched (using the browswer) and didn't find it. I believe you though. 😁